modules/tensor_mechanics/src/utils/GuaranteeConsumer.C

```diff
@@ -0,0 +1,65 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "GuaranteeConsumer.h"
+#include "GuaranteeProvider.h"
+
+#include "MooseObject.h"
+#include "Material.h"
+#include "MooseMesh.h"
+#include "FEProblemBase.h"
+#include "InputParameters.h"
+#include "BlockRestrictable.h"
+
+GuaranteeConsumer::GuaranteeConsumer(MooseObject * moose_object)
+  : _gc_params(moose_object->parameters()),
+    _gc_feproblem(_gc_params.get<FEProblemBase *>("_fe_problem_base")),
+    _gc_block_restrict(dynamic_cast<BlockRestrictable *>(moose_object))
+{
+}
+
+bool
+GuaranteeConsumer::hasGuaranteedMaterialProperty(const MaterialPropertyName & prop_name,
+                                                 Guarantee guarantee)
+{
+  if (!_gc_feproblem->startedInitialSetup())
+    mooseError("hasGuaranteedMaterialProperty() needs to be called in initialSetup()");
+
+  // Reference to MaterialWarehouse for testing and retrieving block ids
+  const auto & warehouse = _gc_feproblem->getMaterialWarehouse();
+
+  // Complete set of ids that this object is active
+  const auto & ids = (_gc_block_restrict && _gc_block_restrict->blockRestricted())
+                         ? _gc_block_restrict->blockIDs()
+                         : _gc_feproblem->mesh().meshSubdomains();
+
+  // Loop over each id for this object
+  for (const auto & id : ids)
+  {
+    // If block materials exist, look if any declare the elasticity tensor
```

Comment:
This has general applicability beyond the elasticity tensor, so I'd change "elasticity tensor" to "material property"